src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (number) => {
+    if (number !== undefined) {
+      result += number;
+
+      return adder;
+    } else {
+      const tempResult = result;
+
+      result = 0;
+
+      return tempResult;
+    }
```

Comment:
If you are using return statements in if, you could simplify like so:
```suggestion
    if (number !== undefined) {
      result += number;

      return adder;
    } 
      const tempResult = result;
      result = 0;

      return tempResult;
    
```